### ISSUE TITLE
`NoEnvOutsideConfig`: false positives when scanning a Laravel package's own config/ dir without explicit `<configDirectory>`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#940]
## Description

When scanning a Laravel **package** (not an app), the `NoEnvOutsideConfig` rule fires on every `env(...)` call inside the package's own `config/*.php`, even though those files are the canonical, documented place to call `env()`.

This is a regression in user experience from #756 → #858: #858 added the `<configDirectory>` opt-in (good), and the fallback was tightened to `ApplicationProvider::getApp()->configPath()` (good for apps). But for **packages** that fallback resolves to the bundled Orchestra Testbench config dir, never the package's own `config/`, so every `env()` call in the package's config files is reported.

The user has to know to add `<configDirectory name="config"/>` to the plugin config. Without it the diagnostic is wrong, not just imprecise.

## Reproduction

Audited [imdhemy/laravel-in-app-purchases](https://github.com/imdhemy/laravel-in-app-purchases) ([branch update-psalm](https://github.com/imdhemy/laravel-in-app-purchases/tree/update-psalm)) with plugin v4.x on Psalm 6.16.1.

`psalm.xml`:

```xml
<projectFiles>
    <directory name="config"/>
    <directory name="routes"/>
    <directory name="src"/>
    <ignoreFiles><directory name="vendor"/></ignoreFiles>
</projectFiles>

<plugins>
    <pluginClass class="Psalm\LaravelPlugin\Plugin"/>
</plugins>
```

(No `<configDirectory>` element inside `<pluginClass>`.)

Output:

```
ERROR: NoEnvOutsideConfig at config/liap.php:33:35
    'google_play_package_name' => env('GOOGLE_PLAY_PACKAGE_NAME', 'com.some.thing'),
ERROR: NoEnvOutsideConfig at config/liap.php:44:28
    'appstore_password' => env('APPSTORE_PASSWORD', ''),
ERROR: NoEnvOutsideConfig at config/liap.php:97:34
    'appstore_private_key_id' => env('APPSTORE_PRIVATE_KEY_ID'),
ERROR: NoEnvOutsideConfig at config/liap.php:99:31
    'appstore_private_key' => env('APPSTORE_PRIVATE_KEY'),
ERROR: NoEnvOutsideConfig at config/liap.php:101:29
    'appstore_issuer_id' => env('APPSTORE_ISSUER_ID'),
ERROR: NoEnvOutsideConfig at config/liap.php:103:29
    'appstore_bundle_id' => env('APPSTORE_BUNDLE_ID'),
```

The reported file `config/liap.php` is the package's published config, merged at runtime via `LiapServiceProvider::mergeConfigFrom(__DIR__.'/../../config/liap.php', 'liap')`. It is a config file by Laravel convention.

## Root cause

`Plugin::initNoEnvOutsideConfigHandler()` (src/Plugin.php:266-275):

```php
$directories = $pluginConfig->configDirectories;

if ($directories === []) {
    $directories = [ApplicationProvider::getApp()->configPath()];
}
```

`ApplicationProvider::getApp()` returns the plugin's internal Orchestra Testbench Laravel app. Its `configPath()` is `vendor/orchestra/testbench-core/laravel/config`, not the analysed package's `config/`. For Laravel **packages** the fallback is therefore guaranteed to miss the package's own config directory.

## Proposed fix

Two options, in increasing order of work:

### Option A (minimal): include the project root's `config/` in the fallback

When `<configDirectory>` is empty, fall back to **both**:

1. `ApplicationProvider::getApp()->configPath()` (current behaviour, useful for projects without a top-level `config/`).
2. `getcwd() . '/config'` if it exists.

`getcwd()` is the project root during Psalm runs; Psalm's own config resolution uses the same anchor. This catches every package and every app that follows the Laravel convention without any user action.

### Option B (more accurate): discover config files from `mergeConfigFrom()` / `publishes()` calls

Scan service providers for `mergeConfigFrom(__FILE_PATH__, ...)` and `publishes([... => config_path(...)], 'config')`, resolve `__DIR__`-relative paths at scan time, and add the parent directories to the configDirectory list. This matches the actual semantic ("file is a Laravel config file") rather than the conventional path layout. It also handles non-standard layouts (e.g. `resources/config/`).

Option B is more correct but requires a scanner pass; Option A is a 3-line change in `Plugin.php`.

### Either way

Detect the "package, no `<configDirectory>` set, fallback resolved to zero project-relevant dirs" case explicitly and emit a one-line `$progress->info()` pointing to the [`configDirectory` docs](https://psalm.github.io/psalm-plugin-laravel/config/#configdirectory). The current warning only fires when **all** inputs fail to resolve, which never happens for the testbench fallback.

## Impact

Affects every Laravel package analysed with the plugin that ships a `config/*.php` and does not know to set `<configDirectory>`. In one-shot audits across `spatie/*`, `imdhemy/*`, `laravel-notification-channels/*` packages, this is a consistent stream of false positives that pushes maintainers to suppress the rule entirely rather than fix the (real) `env()` calls outside config.

Documentation alone is not enough here — the fallback path encodes "this is a Laravel app", but the plugin is also widely used on Laravel packages. The default should not silently produce wrong results on the second case.

## Workaround (for users hitting this today)

Add to `psalm.xml`:

```xml
<pluginClass class="Psalm\LaravelPlugin\Plugin">
    <configDirectory name="config"/>
</pluginClass>
```

**Comment by alies-dev** [comment]
fixed by https://github.com/psalm/psalm-plugin-laravel/pull/949
